keycloak/keycloak-admin-ui
# Realm roles(role attributes): adds initial empty field on load, also adds empty field on save

## Motivation

#299, #291
https://issues.redhat.com/browse/APPDUX-772

## Brief Description


This PR allows a user to successfully save, update, and delete role attributes.

The work done in this issue addresses the observation by @ssilvert that we always need to have an empty field at the bottom of the field array so a user is able to easily delete the last key/value pair in the list without having to add an additional row. 

@edewit and I also made a large effort to address #291 in this PR. We spent a great deal of time attempting many work-arounds however due to some bugs in react-hook-form this is not possible with their current API. Therefore at the moment all key/value pairs can be edited after `save`, and the last (empty) input field has functionality for both adding and deleting the row. 


## Verification Steps



1. Go to Realm Roles section.
2. Click on an existing role.
3. Go to the "Attributes" tab.
4. Try adding an attribute and saving.
5. Try adding multiple attributes and saving.
6. Try deleting one or multiple attributes and saving.
8. Navigate to a different page in the UI, and then navigate back to "Attributes" to confirm the data was saved successfully.

## Checklist:


## Additional Notes
